nym-vpn-core/crates/nym-vpnd/src/service/vpn_service.rs

```diff
@@ -1017,6 +1071,157 @@ impl NymVpnService {
         });
     }
 
+    async fn handle_enable_socks5(
```

Comment:
These `handle_xxx()` functions are generally used to save to the configuration with the idea that when the daemon starts it knows what settings to use and can connect even before the app has started.

So the configuration needs to be updated (to v3) and it needs to hold the new socks5 configuration options.  When the daemon connects/disconnects the tunnel it will also start/stop the socks5 listener.